src/runtime/linux_host_cpu_count.cpp

```diff
@@ -3,8 +3,27 @@
 extern "C" {
 
 extern long sysconf(int);
+extern int sched_getaffinity(int32_t pid, size_t cpusetsize, unsigned long *cpuset);
 
 WEAK int halide_host_cpu_count() {
-    return sysconf(84);
-}
+    unsigned long count = 0, cpuset[8];  // enough bits for 256 CPUs on 32 bit, or 512 CPUs on 64 bit
```

Comment:
I thought a long was always 32-bits. Please just use a uint32_t to make it explicit, and change the declared signature of sched_getaffinity to match. The cpusetsize arg is measured in bytes anyway.

Reply:
Done.

A long is always 32 bits on Windows, but not Linux.  https://en.wikipedia.org/wiki/64-bit_computing#64-bit_data_models
 (LLP64 vs LP64)

Reply:
Good to know. The runtime module is compiled for a generic OS, not linux, so we should definitely avoid any types that have OS-dependent sizes.

Reply:
(despite being linux-specific code)